### ISSUE TITLE
Feature: ldap search scope

### ISSUE THREAD

**Issue by ds-04** [ubccr/coldfront#629]
### Description

It appears the ldap search scope is ``one``, it would be useful if it could be ``sub``.

### Component

Other

**Comment by ds-04** [comment]
Certainly auth is using ``onelevel``  via ``python-ldap``, changing to be able to use ``subtree`` would be good.

```
coldfront/config/plugins
diff ldap.py ldap.py.orig

32,33c23,24
< AUTH_LDAP_USER_SEARCH = LDAPSearch(AUTH_LDAP_USER_SEARCH_BASE, ldap.SCOPE_SUBTREE, '(uid=%(user)s)')
< AUTH_LDAP_GROUP_SEARCH = LDAPSearch(AUTH_LDAP_GROUP_SEARCH_BASE, ldap.SCOPE_SUBTREE, '(objectClass=groupOfNames)')
---
> AUTH_LDAP_USER_SEARCH = LDAPSearch(AUTH_LDAP_USER_SEARCH_BASE, ldap.SCOPE_ONELEVEL, '(uid=%(user)s)')
> AUTH_LDAP_GROUP_SEARCH = LDAPSearch(AUTH_LDAP_GROUP_SEARCH_BASE, ldap.SCOPE_ONELEVEL, '(objectClass=groupOfNames)')
```


----------

The ldap user search I think is actually using ``subtree`` by default (``python ldap3`` default) and there is an issue with one of my test accounts in the test OpenLDAP instance in a particular OU, which I'm trying to get to the bottom of.

**Comment by aebruno** [comment]
Agree, this looks good. Probably makes sense to make this a config option?

**Comment by ds-04** [comment]
> Agree, this looks good. Probably makes sense to make this a config option?

Yes it would make sense as a config option.

**Comment by ds-04** [comment]
PR #637 created for review, thanks

**Comment by ds-04** [comment]
was resolved by https://github.com/ubccr/coldfront/pull/641 ... closing this
